src/components/Message.vue

```diff
@@ -232,6 +233,7 @@ export default {
 		},
 		onToggleSeen() {
 			this.$store.dispatch('toggleEnvelopeSeen', this.envelope)
+			emit('mail:interaction', {type: 'message-read', recipient: this.selectedUser.user})
```

Comment:
again. use `this.$store.dispatch(...).then(() => { ... })` to await the store action